File: hotel-smart-booking-backend/src/main/java/com/example/hotelsmartbookingbackend/service/RefundService.java
package com.example.hotelsmartbookingbackend.service;

import com.example.hotelsmartbookingbackend.dto.request.ApproveRefundRequest;
import com.example.hotelsmartbookingbackend.dto.request.RefundRequest;
import com.example.hotelsmartbookingbackend.dto.request.RejectRefundRequest;
import com.example.hotelsmartbookingbackend.dto.response.CustomerRequestResponse;

import java.util.List;

public interface RefundService {
    CustomerRequestResponse submitRefundRequest(Integer bookingId, RefundRequest request, String customerEmail);
    CustomerRequestResponse approveRefundRequest(Integer requestId, ApproveRefundRequest request, String staffEmail);
    CustomerRequestResponse rejectRefundRequest(Integer requestId, RejectRefundRequest request, String staffEmail);
    List<CustomerRequestResponse> getPendingRefundRequests(String staffEmail);
}

File: hotel-smart-booking-frontend/src/services/refundService.js
import axiosInstance from './axiosInstance';

export const submitRefundRequest = async (bookingId, reason) => {
  const response = await axiosInstance.post(`/bookings/${bookingId}/refund-request`, { reason });
  return response.data;
};

export const getPendingRefundRequests = async () => {
  const response = await axiosInstance.get('/refund-requests/pending');
  return response.data;
};

export const approveRefundRequest = async (requestId, refundAmount) => {
  const body = refundAmount ? { refundAmount } : {};
  const response = await axiosInstance.post(`/refund-requests/${requestId}/approve`, body);
  return response.data;
};

export const rejectRefundRequest = async (requestId, rejectionReason) => {
  const response = await axiosInstance.post(`/refund-requests/${requestId}/reject`, { rejectionReason });
  return response.data;
};
